### PR TITLE
Update AWS EKS to 1.23 along with a bunch of chart upgrades

### PR DESCRIPTION
**Pull request by DavidSpek**
## Summary
This PR updates the EKS control plane to version 1.23, along with updates for the EKS addons. Also, a bunch of helm charts in bootstrap are updated and the image vendoring settings have been updated and improved.

## Test Plan
Local linking

## Checklist

- [ ]  No images hosted from dockerhub
- [ ]  Are dashboards present to understand the health of the application.  There **must** be at least 1 of these
    - [ ]  all databases should have dashboards
    - [ ]  ideally also have at least cpu/mem utilization dashboards for webserver tier of the app
    - [ ]  you can use `plural from-grafana` to convert a grafana dashboard found via google to our CRD
- [ ]  Are scaling runbooks present
    - [ ]  all databases **must** have scaling runbooks
    - [ ]  you can use the charts in `pluralsh/module-library` to accelerate this
- [ ]  do you need to add config overlays?
    - [ ]  inputing secrets
    - [ ]  configuring autoscaling
- [ ]  If there’s a web-facing component to the app, we need to support OIDC authentication and setting up private networks if no authentication option is viable
- [ ]  All major clouds must be supported
    - [ ]  Azure
    - [ ]  AWS
    - [ ]  GCP

### CODE REVIEW

File: bootstrap/terraform/aws-bootstrap/deps.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: plural.sh/v1alpha1
 kind: Dependencies
 metadata:
   description: Creates an EKS cluster and prepares it for bootstrapping 
-  version: 0.1.49
+  version: 0.1.50
```
[the remark this answers is not in the document]
Comment (DavidSpek) [reply bootstrap/terraform/aws-bootstrap/deps.yaml:5]: It's the node upgrades that are breaking, which will be done in a follow up PR.

File: bootstrap/terraform/aws-bootstrap/aws-lb-controller.tf [hunk 2]
```diff
@@ -162,7 +174,13 @@ resource "aws_iam_policy" "alb" {
                 "arn:aws:elasticloadbalancing:*:*:targetgroup/*/*",
                 "arn:aws:elasticloadbalancing:*:*:loadbalancer/net/*/*",
                 "arn:aws:elasticloadbalancing:*:*:loadbalancer/app/*/*"
-            ]
+            ],
+            "Condition": {
```
Comment (michaeljguarino) [inline bootstrap/terraform/aws-bootstrap/aws-lb-controller.tf:178]: I remember this causing the controller to break around product hunt, unless we're confident it's kosher, we should revert it

File: bootstrap/terraform/aws-bootstrap/aws-lb-controller.tf [hunk 3]
```diff
@@ -196,6 +214,28 @@ resource "aws_iam_policy" "alb" {
...
+            "Effect": "Allow",
+            "Action": [
+                "elasticloadbalancing:AddTags"
+            ],
+            "Resource": [
+                "arn:aws:elasticloadbalancing:*:*:targetgroup/*/*",
+                "arn:aws:elasticloadbalancing:*:*:loadbalancer/net/*/*",
+                "arn:aws:elasticloadbalancing:*:*:loadbalancer/app/*/*"
+            ],
+            "Condition": {
```
Comment (michaeljguarino) [inline bootstrap/terraform/aws-bootstrap/aws-lb-controller.tf:227]: Same here, I'm concerned this has the same problems w/ borked iam perms

File: bootstrap/terraform/aws-bootstrap/deps.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaeljguarino) [reply bootstrap/terraform/aws-bootstrap/deps.yaml:5]: kk cool

File: bootstrap/terraform/aws-bootstrap/aws-lb-controller.tf (hunk 3 above)
Comment (DavidSpek) [reply bootstrap/terraform/aws-bootstrap/aws-lb-controller.tf:227]: This is the new upstream IAM permission that includes the fix for the tags. I just tested with creating a LoadBalancer service and it worked as expected.

File: bootstrap/terraform/aws-bootstrap/aws-lb-controller.tf (hunk 2 above)
Comment (DavidSpek) [reply bootstrap/terraform/aws-bootstrap/aws-lb-controller.tf:178]: That was with the tags not being able to be added. which now has a new policy entry below.

File: bootstrap/terraform/aws-bootstrap/aws-lb-controller.tf (hunk 3 above)
Comment (michaeljguarino) [reply bootstrap/terraform/aws-bootstrap/aws-lb-controller.tf:227]: ok awesome

**Review by michaeljguarino — APPROVED** [review]

**Pull request merged**